File: frontend-biblio/app/src/main/java/com/fex/projetandroidm1/model/Livre.java
package com.fex.projetandroidm1.model;

import org.json.JSONException;
import org.json.JSONObject;

public class Livre {

    private String numlivre;
    private String designlivre;
    private String autlivre;
    private String dateeditlivre;
    private String dispolivre;

    public Livre() {
    }

    public Livre(String numlivre, String designlivre, String autlivre, String dateeditlivre, String dispolivre) {
        this.numlivre = numlivre;
        this.designlivre = designlivre;
        this.autlivre = autlivre;
        this.dateeditlivre = dateeditlivre;
        this.dispolivre = dispolivre;
    }

    public String getNumlivre() {
        return numlivre;
    }

    public void setNumlivre(String numlivre) {
        this.numlivre = numlivre;
    }

    public String getDesignlivre() {
        return designlivre;
    }

    public void setDesignlivre(String designlivre) {
        this.designlivre = designlivre;
    }

    public String getAutlivre() {
        return autlivre;
    }

    public void setAutlivre(String autlivre) {
        this.autlivre = autlivre;
    }

    public String getDateeditlivre() {
        return dateeditlivre;
    }

    public void setDateeditlivre(String dateeditlivre) {
        this.dateeditlivre = dateeditlivre;
    }

    public String getDispolivre() {
        return dispolivre;
    }

    public void setDispolivre(String dispolivre) {
        this.dispolivre = dispolivre;
    }
}

File: backend-biblio/src/Entity/Livre.php
<?php

namespace App\Entity;

use ApiPlatform\Core\Annotation\ApiResource;
use App\Repository\LivreRepository;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ApiResource()
 * @ORM\Entity(repositoryClass=LivreRepository::class)
 */
class Livre
{
    /**
     * @ORM\Id
     * @ORM\Column(type="string", length=25)
     */
    private $numlivre;
    /**
     * @ORM\Column(type="string", length=50)
     */
    private $designlivre;
    /**
     * @ORM\Column(type="string", length=30)
     */
    private $autlivre;
    /**
     * @ORM\Column(type="string", length=25)
     */
    private $dateeditlivre;
    /**
     * @ORM\Column(type="string", length=5)
     */
    private $dispolivre;

    public function getNumlivre(): ?string
    {
        return $this->numlivre;
    }
    public function setNumlivre($numlivre)
    {
        $this->numlivre = $numlivre;
    }
    public function getDesignlivre(): ?string
    {
        return $this->designlivre;
    }
    public function setDesignlivre($designlivre)
    {
        $this->designlivre = $designlivre;
    }
    public function getAutlivre(): ?string
    {
        return $this->autlivre;
    }
    public function setAutlivre($autlivre)
    {
        $this->autlivre = $autlivre;
    }
    public function getDateeditlivre(): ?string
    {
        return $this->dateeditlivre;
    }
    public function setDateeditlivre($dateeditlivre)
    {
        $this->dateeditlivre = $dateeditlivre;
    }
    public function getDispolivre(): ?string
    {
        return $this->dispolivre;
    }
    public function setDispolivre($dispolivre)
    {
        $this->dispolivre = $dispolivre;
    }
}
